### PR TITLE
Side 980 whisper ldk new list whisper type

### PR DESCRIPTION
**Pull request by BillyMurrayOlive**
Go LDK
- Fixed panic if *proto.session was nil pointer. NewSessionFromProto now returns an error and is handled gracefully in the service servers.
- Added new whisper type

Node LDK
- [x] Add new whisper type "list".

C# LDK
- [x] Add new whisper type "list".

### CODE REVIEW

**Review by scottkipferOlive — APPROVED** [review]

**Comment by BillyMurrayOlive** [comment]
This last round of commits fixed not being able to identify the list element type once it was marshaled. I added custom MarshalJSON methods for each element type that add a "type" field to the JSON. Also added a test to ensure this is working in the future.

This issue also exists with the form whisper type. So I am going to fix it now, improve the whisper-form example to include all types, and add some test for marshaling that type.

**Comment by BillyMurrayOlive** [comment]
- [ ] Need to apply these recent changes to the Go LDK over to the Node LDK.

@richseviora You mentioned that you were willing to take on the C# implementation here. Do you want to do that as part of this PR, or would you rather handle it separately?

**Review by scottkipferOlive — APPROVED** [review]

File: ldk/node/src/hostClients/whisperService.ts [hunk 1]
```diff
@@ -108,6 +108,88 @@ export interface WhisperFormConfig extends Whisper {
...
+  extra?: boolean;
+  /**
+   *  Value the UI uses to order the form inputs.
+   *  Value must be greater than 0
+   *  If this value is ommited it will default to 0
+   */
+  order?: number;
+}
+
+// eslint-disable-next-line no-shadow
```
Comment (BillyMurrayOlive) [inline ldk/node/src/hostClients/whisperService.ts:129]: - [ ] @richseviora You asked me to make a note to remind you to look at this shadow issue and the one for WhisperListAlign as well.

**Comment by BillyMurrayOlive** [comment]
Rich and I talked and he graciously offered to take over the C# portion of this. So this PR will be on hold until that work is complete.

**Review by JonBernardOlive — APPROVED** [review]

**Review by iangallOlive — APPROVED** [review]

**Pull request merged**